### PR TITLE
initial implementation of 'rake publish' command

### PR DESCRIPTION
**Pull request by subelsky**

### CODE REVIEW

File: publisher/config/environments/test.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+Rails.application.configure do
```
Comment (jonathanroehm) [inline publisher/config/environments/test.rb:1]: do we need a test environment for this app? are there tests?

File: publisher/Rakefile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      return true # exit status 1 means no files found, which is the success condition for this method
+    else
+      abort "Received unexpected error while checking for pre-existing files in '#{destination_uri}': #{res.exitstatus}.  Please try again."
+      break
+    end
+  end
+end
+
+desc 'Compile nfg_ui assets and upload to S3 for public access'
+task :publish, %i[nfg_ui_version override_mode] => %i[assets:precompile] do |_, nfg_ui_version: nil, override_mode: nil|
```
Comment (jonathanroehm) [inline publisher/Rakefile:37]: Can you confirm the order of operations for `publish` to fire? Would we release first then publish?

File: publisher/config/environments/test.rb (hunk 1 above)
Comment (subelsky) [reply publisher/config/environments/test.rb:1]: it's not really a standalone app, so I couldn't write normal tests. though I could write a test to exercise the release code. then the rake task would just be calling a PORO that does the shell stuff.

File: publisher/Rakefile (hunk 2 above)
Comment (subelsky) [reply publisher/Rakefile:37]: normally you would just call 'rake release' which would take care of invoking 'rake publish'. but if for some reason you wanted to publish as a separate step (like to correct an error) that could be done inside the publisher/ app using 'rake publish'

Comment (jonathanroehm) [reply publisher/Rakefile:37]: I'd be in favor of having at least a basic test here that evaluates your error messages and statuses etc.

File: publisher/config/environments/test.rb (hunk 1 above)
Comment (jonathanroehm) [reply publisher/config/environments/test.rb:1]: I'd be in favor of having at least a basic test here that evaluates your error messages and statuses etc. (Edit: moved to this comment)

File: publisher/Rakefile (hunk 2 above)
Comment (jonathanroehm) [reply publisher/Rakefile:37]: Awesome! Love it.

Comment (jonathanroehm) [reply publisher/Rakefile:37]: @subelsky would you update the Readme doc to include step by step instructions to publish in the event of an error as part of this PR? You can add a new section above "Usage"

**Comment by subelsky** [comment]
@jonathanroehm tests and documentation have been added, please let me know if there's anything else I can do to make this user-friendly/error-resistant

**Review by jonathanroehm — APPROVED** [review]
Awesome work! Thanks for the specs and docs.

**Pull request merged**